capi/src/write/sitemap.ts

```diff
@@ -0,0 +1,42 @@
+import * as t from "../types";
+import * as path from "path";
+import * as fs from "fs-extra";
+
+const now = new Date();
+const yyyyMmDd = `${now.getFullYear()}-${now.getMonth()}-${now.getDate()}`;
+
+function createSitemapRoute(route: string): string {
+  return ` <url>
+    <loc>http://docs.amplify.aws${route}</loc>
+    <lastmod>${yyyyMmDd}</lastmod>
+    <changefreq>hourly</changefreq>
+    <priority>.5</priority>
+  </url>`;
+}
+
+export async function sitemap(config: t.Config, ctx: t.Ctx): Promise<void> {
+  const routes = new Map<string, true>();
+
+  for (const [, page] of ctx.pageBySrcPath.entries()) {
+    routes.set(page.route, true);
+    if (page.versions) {
+      Object.entries(page.versions).forEach(([, version]) =>
+        routes.set(version, true),
+      );
+    }
+  }
+
+  const src = `<?xml version="1.0" encoding="utf-8"?>
+<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
+  ${[...routes.keys()].map(createSitemapRoute).join("\n")}
+</urlset>
+`;
+
+  const destPathPieces = config.outDir.split(path.sep);
```

Comment:
If something in the route assignment broke it seems we would just generate an empty sitemap file. Should we throw some kind of error instead?

Reply:
Maybe I don't understand correctly...

destPathPieces being of length 0 won't happen with the Amplify Console environment... and if anyone is running Capi locally, and they have the project at their filesystem root (before library, before user)... then they're begging for trouble.

Reply:
Moved this into an optional util in the config